src/CanonicalizeRequest.jl

```diff
@@ -0,0 +1,33 @@
+module CanonicalizeRequest
+
+import ..Layer, ..request
+using ..Messages
+using ..Strings.tocameldash!
+
+
+"""
+    request(CanonicalizeLayer, method, ::URI, headers, body) -> HTTP.Response
+
+Rewrite request and response headers in Canonical-Camel-Dash-Format.
+"""
+
+abstract type CanonicalizeLayer{Next <: Layer} <: Layer end
+export CanonicalizeLayer
+
+function request(::Type{CanonicalizeLayer{Next}},
+                 method::String, uri, headers, body; kw...) where Next
```

Comment:
why note `method::Method` here? in the old client.jl file, we converted from a potential `String` method argument to the enum pretty quickly.

Reply:
The enum is not used anywhere except handlers.jl and on one line in a compatibility shim in client.jl.